### PR TITLE
Feature/add tests coverage al

### PR DESCRIPTION
**Pull request by alexnetflix**
This PR adds test coverage reports and does some minor cleanup on readme page(more badges etc)

@cyzhao @apanicker-nflx  I'd suggest to merge it to both 1.12 and 2.0 for comparison on test coverage moving forward.

See how it would look like:
screen shot 2018-10-02 at 10 50 23 am

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+codecov:
+  branch: master
```
Comment (apanicker-nflx) [inline .codecov.yml]: can probably remove this as it defaults to `master`

File: .codecov.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+  # strict_yaml_branch: master # Enable this if we want to use the yml file in master to dictate the reports for all branches
```
Comment (apanicker-nflx) [inline .codecov.yml]: Maybe a good idea to uncomment this so that the yaml file is restricted to master.

File: build.gradle [hunk 3]
```diff
@@ -19,41 +20,109 @@ ext.githubProjectName = rootProject.name // Change if github project name is not
...
 }
+
+
+
+/**********************************
+ * Coverage Tasks
+ **********************************/
+task codeCoverageReport(type: JacocoReport, group: "Coverage reports") {
+    executionData fileTree(project.rootDir.absolutePath).include("**/build/jacoco/*.exec")
+    //dependsOn subprojects*.test
```
Comment (apanicker-nflx) [inline build.gradle]: remove commented code

File: build.gradle [hunk 4]
```diff
@@ -19,41 +20,109 @@ ext.githubProjectName = rootProject.name // Change if github project name is not
 apply plugin: 'project-report'
 apply from: "$rootDir/versionsOfDependencies.gradle"
 
+allprojects {
+    apply plugin: 'idea'
+    apply plugin: 'jacoco'
+    apply plugin: 'eclipse'
```
Comment (falu2010-netflix) [inline build.gradle]: why do we need eclipse and idea plugin?

File: README.md [hunk 5]
```diff
@@ -67,18 +81,3 @@ com.netflix.conductor.dao.RedisWorkflowModule
 ## Get Support
 Conductor is maintained by Media Workflow Infrastructure team at Netflix.  Use github issue tracking for any support request.  
 
-## LICENSE
```
Comment (falu2010-netflix) [inline README.md]: Why is the license being removed?

File: build.gradle (hunk 4 above)
Comment (alexnetflix) [reply build.gradle]: we don't but I assume other users might use both

File: README.md (hunk 5 above)
Comment (alexnetflix) [reply README.md]: it's a badge(link) now

**Comment by cyzhao** [comment]
@alexnetflix we only need to merge this dev, 2.0 branch is now deprecated. Please resolve the conflict then merge to dev.

**Pull request merged**